lib/transformations.js

```diff
@@ -1,38 +1,65 @@
 'use strict'
+function vertical (matrix) {
+  return matrix.slice().reverse()
+}
 
- exports.flip = function (matrix, direction) {
-  if (!['none', 'horizontal', 'vertical', 'both'].includes(direction)) {
-    throw new Error('You can only rotate by 0, 90, 180, or 270 degrees')
-  }
-
-  function vertical (matrix) {
-    return matrix.reverse()
-  }
+function horizontal (matrix) {
+  return matrix.map(r => r.reverse() )
+}
 
-  function horizontal (matrix) {
-    return matrix.map(r => r.reverse() )
+function flip (matrix, direction) {
```

Comment:
I never enjoyed this block of code anyway. I might suggest a hash:

```suggestion
function flip (matrix, direction) {

  const matrixFunctions = {
    none: m => m,
    vertical: m => vertical(m),
    horizontal: m => horizontal(m),
    both: m => vertical(horizontal(m))
  }
  
  const flipFunction = matrixFunctions[direction]
  
  if (flipFunction) {
    return flipFunction(matrix)
  }
  
  throw new Error('You can only flip by none, horizontal, vertical, or both')
}
```

Reply:
Please review entire transformation.js file.  It did come out cleaner.  

As a matter of choice, always opt for putting the hash outside of the function to prevent the hash entry functions from having to be compiled each time the function is called.